File: arrays/plusOne.java
package arrays;

import java.util.Arrays;
import java.util.LinkedList;

public class plusOne {
    public static void main(String[] args) {
        SolutionPO solution = new SolutionPO();
        System.out.println(Arrays.toString(solution.plusOne(new int[]{9})));
        System.out.println(Arrays.toString(solution.plusOne(new int[]{1,2,3})));
        System.out.println(Arrays.toString(solution.plusOne(new int[]{4,3,2,1})));

    }
    
}

class SolutionPO {
    public int[] plusOne(int[] digits) {
        LinkedList <Integer> result = new LinkedList<Integer>();
        int n = digits.length-1;
        int k =1;
        while(k>0 || n>=0){
            int add = (n>=0)? digits[n]+k: k;
            result.addFirst(add %10);
            k = add /10;
            n--;
        }
        
        int [] res = new int[result.size()];
        for(int i = 0 ; i< res.length; i++){
            res[i] = result.get(i);
        }
        return res;
        
    }
}

File: arrays/plusOne.py
# https://leetcode.com/problems/plus-one/

class Solution:
    # TC: O(N)
    # SC: O(N)
    def plusOne(self, digits: List[int]) -> List[int]:
        
        n = len(digits)-1
        add = 1

        while n >= 0:
            addition = digits[n]+ add
            if addition <= 9:
                digits[n] = addition
                return digits
            else:
                digits[n] = addition % 10
                add = addition // 10
            
            n-=1

        return ([add] if add else []) + digits



        
